api/server/middleware/debug.go

```diff
@@ -54,13 +54,22 @@ func DebugRequestMiddleware(handler func(ctx context.Context, w http.ResponseWri
 	}
 }
 
-func maskSecretKeys(inp interface{}) {
+func maskSecretKeys(inp interface{}, path string) {
+	// Remove any query string from the path
+	idx := strings.Index(path, "?")
```

Comment:
nit: this would still fail if an URL with an anchor is passed (`"/v1.30/secrets/create#some-fragment?key=val"`)

Perhaps as a follow-up, something like;

```go
if u, err := url.Parse(path); err == nil {
	path = u.Path
}
```